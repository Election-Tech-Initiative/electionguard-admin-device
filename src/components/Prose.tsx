import styled from 'styled-components'

interface Props {
  textCenter?: boolean
  compact?: boolean
}

const Prose = styled('div')<Props>`
  margin: ${({ textCenter }) => (textCenter ? 'auto' : undefined)};
  width: 100%;
  max-width: 66ch;
  text-align: ${({ textCenter }) => (textCenter ? 'center' : undefined)};
  line-height: 1.2;
  @media (min-width: 480px) {
    line-height: 1.3;
  }
  & h1 {
    margin: 2rem 0 1rem;
    line-height: 1.1;
    font-size: 1.5rem;
  }
  & h2 {
    margin: 1.5rem 0 0.75rem;
    font-size: 1.25rem;
  }
  & h3,
  & p {
    margin-top: ${({ compact }) => (compact ? '0' : '1rem')};
    margin-bottom: ${({ compact }) => (compact ? '0' : '1rem')};
    font-size: 1rem;
  }
  & h1 + h2 {
    margin-top: -0.75rem;
  }
  & h1 + p,
  & h2 + p {
    margin-top: -0.75rem;
  }
  & h3 + p {
    margin-top: ${({ compact }) => (compact ? 0 : '-1rem')};
  }
  & :first-child {
    margin-top: 0;
  }
  & :last-child {
    margin-bottom: 0;
  }
`

export default Prose
