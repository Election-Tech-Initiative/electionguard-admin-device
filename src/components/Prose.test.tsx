import React from 'react'
import { render } from '@testing-library/react'

import Prose from './Prose'

const proseContent = (
  <>
    <h1>Heading 1</h1>
    <h2>Heading 2</h2>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam
      modi magnam neque. Molestias recusandae, officia maiores nam pariatur
      earum qui inventore minus enim adipisci nemo voluptate at harum?
    </p>
    <h1>Heading 1</h1>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam
      modi magnam neque. Molestias recusandae, officia maiores nam pariatur
      earum qui inventore minus enim adipisci nemo voluptate at harum?
    </p>
    <h2>Heading 2</h2>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam
      modi magnam neque. Molestias recusandae, officia maiores nam pariatur
      earum qui inventore minus enim adipisci nemo voluptate at harum?
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam
      modi magnam neque. Molestias recusandae, officia maiores nam pariatur
      earum qui inventore minus enim adipisci nemo voluptate at harum?
    </p>
    <h3>Heading 3</h3>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam
      modi magnam neque. Molestias recusandae, officia maiores nam pariatur
      earum qui inventore minus enim adipisci nemo voluptate at harum?
    </p>
  </>
)

it('renders Prose defaults', async () => {
  const { container } = render(<Prose>{proseContent}</Prose>)
  expect(container.firstChild).toMatchSnapshot()
})

it('renders Prose with compact spacing', async () => {
  const { container } = render(<Prose compact>{proseContent} </Prose>)
  expect(container.firstChild).toMatchSnapshot()
})

it('renders Prose with compact spacing', async () => {
  const { container } = render(<Prose textCenter>{proseContent}</Prose>)
  expect(container.firstChild).toMatchSnapshot()
})
