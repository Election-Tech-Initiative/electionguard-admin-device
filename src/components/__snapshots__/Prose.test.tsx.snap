// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Prose defaults 1`] = `
<div
  class="sc-bdVaJa fUpyWZ"
>
  <h1>
    Heading 1
  </h1>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h1>
    Heading 1
  </h1>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h3>
    Heading 3
  </h3>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
</div>
`;

exports[`renders Prose with compact spacing 1`] = `
<div
  class="sc-bdVaJa hMDJJy"
>
  <h1>
    Heading 1
  </h1>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h1>
    Heading 1
  </h1>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h3>
    Heading 3
  </h3>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
   
</div>
`;

exports[`renders Prose with compact spacing 2`] = `
<div
  class="sc-bdVaJa UBVRg"
>
  <h1>
    Heading 1
  </h1>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h1>
    Heading 1
  </h1>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h2>
    Heading 2
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
  <h3>
    Heading 3
  </h3>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor omnis quam modi magnam neque. Molestias recusandae, officia maiores nam pariatur earum qui inventore minus enim adipisci nemo voluptate at harum?
  </p>
</div>
`;
